### PR TITLE
zero-install setup

### PR DESCRIPTION
**Pull request by jenbutongit**
# Description

tl;dr: Smoke tests 6m -> 3m _if there are no changes to dependencies(?)*_ build-tests should cap around 5ish minutes. Previously could reach 8-10min

~`* `Not sure why the below took 13 minutes. This might be a docker caching issue~
If the docker caching is only to circumvent the yarn fetching issue, we can remove it 

At the moment, our CI/CD pipelines are bottlenecked by yarn install. (would be the same with npm install). Even if we cached .yarn/cache on the pipeline, that would mean downloading the cache which is slower on these executors. same with node_modules.

With yarn 2, you can opt-in to a zero-install philosophy. Yarn 2 caches in the project directory, not in a global dir, so you pull the project, run yarn install (you don't fetch anything because it's all there) and you're off. It takes the 2-3 minutes yarn install fetching down to 30-40s.

To prevent bad actors, if you are not part of the maintainers team, we will run yarn install --check-cache which does the checksum and pulls any dependencies that are bad

reference:
https://yarnpkg.com/features/zero-installs

### changes 

- Commit yarn/cache
- Documentation changes

**CircleCI changes**
- Removing all caching of dependencies on circleCI 
- Updating circleci images (they are now using cimg) which are optimised for CI/CD.
- Removing unused executors 
- Removing reference to circle_trigger (but not the script itself if we need to use it down the line)
- Running yarn install or yarn install --check-cache if users are not part of the maintainers team. 
  - I have set up a circleCI context, which is their way of adding user groups with different security settings. It is hooked into GitHub, and is set up so that any member of maintainers does not need the cache to be checked.
- Model is built during the yarn install step. This is because persisting the workspace adds an extra ~40-60 seconds to the build and will accumulate. 
- Everything is built and tested in parallel (except model build) 

**Smoke test changes**
- use nanoid for smoke tests instead of stringified date

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- [x] Everything runs ok!

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts
- [x] I have checked deployments are working in all environments
- [x] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Comment by RichardBray** [comment]
Who such a mega PR 😱  😱  😱 
I'd love to know more about what this is doing.

**Comment by jenbutongit** [comment]
@RichardBray ha, sorry, I haven't actually written up the PR but needed to see if it worked.

At the moment, our CI/CD pipelines are bottlenecked by yarn install. (would be the same with npm install). Even if we cached .yarn/cache on the pipeline, that would mean downloading the cache which is slower on these executors. same with node_modules. 

With yarn 2, you can opt-in to a zero-install philosophy. Yarn 2 caches in the project directory, not in a global dir, so you pull the project, run `yarn install` (you don't fetch anything because it's all there) and you're off. It takes the 2-3 minutes yarn install fetching down to 30-40s. 

To prevent bad actors, if you are not part of the maintainers team, we will run `yarn install --check-cache` which does the checksum and pulls any dependencies that are _bad_

**Comment by RichardBray** [comment]
> https://yarnpkg.com/features/zero-installs

Ah cool, that sounds really interesting

**Comment by bertrandmc** [comment]
Please add the `do not review` label when appropriate.

**Comment by jenbutongit** [comment]
It's ready for review!!

**Comment by jenbutongit** [comment]
closing in favour of #279 🎻
